### PR TITLE
[wasm] Refactor name convention

### PR DESCRIPTION
**Pull request by ilonatommy**
First batch of TS refactoring. It is restricted to files connected with threading.
- Some names e.g. `mono_log_debug` used in threading are widely used in other locations so were left temporarily unchanged
- Functions exported to C will ~~be translated from one convention to the other, see: `monoWasmEventLoopHasUnsettledInteropPromises`. Is it a good approach?~~ will keep their C-style names for consistency.

### CODE REVIEW

File: src/mono/wasm/runtime/exports-linker.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ const mono_wasm_threads_exports = !MonoWasmThreads ? undefined : {
     // mono-threads-wasm.c
     mono_wasm_pthread_on_pthread_attached,
     // threads.c
-    mono_wasm_eventloop_has_unsettled_interop_promises,
+    mono_wasm_eventloop_has_unsettled_interop_promises: monoWasmEventLoopHasUnsettledInteropPromises,
```
Comment (pavelsavara) [inline src/mono/wasm/runtime/exports-linker.ts:38]: I would prefer to keep the names consistent across languages. 
so rename also C code of `mono_wasm_eventloop_has_unsettled_interop_promises` or keep the original name.

The important thing here is that fulltext search should find both, usage and definitionin both languages.

Comment (ilonatommy) [reply src/mono/wasm/runtime/exports-linker.ts:38]: I am voting for keeping the original name then.

Comment (pavelsavara) [reply src/mono/wasm/runtime/exports-linker.ts]: in this case it's C import. should the same rule apply ? Or the other ?

Comment (maraf) [reply src/mono/wasm/runtime/exports-linker.ts]: I would keep even imports to C in C style convention

Comment (pavelsavara) [reply src/mono/wasm/runtime/exports-linker.ts]: ok

**Comment by ilonatommy** [comment]
Monointerpreter, debugger failures - not connected.

LibraryTests_Threading_Smoke looks like connected, however I cannot reproduce the same failures - I am getting `Arg_PlatformNotSupported` instead. @lambdageek, for the threading sample we:
- build: `build mono+libs -os browser /p:WasmEnableThreads=true`
- run: `dotnet build -bl /p:TargetOS=browser /p:TargetArchitecture=wasm /p:Configuration=Release /t:RunSample .\src\mono\sample\wasm\browser-threads-minimal`
- expect it working without errors in the console, right? I am builing on Windows.

**Comment by pavelsavara** [comment]
build: `/p:MonoWasmBuildVariant=multithread`

**Comment by ilonatommy** [comment]
Text.Json and debugger failures are not related.

**Review by pavelsavara — APPROVED** [review]

**Pull request merged**